Issue 5956 - After an upgrade the server won't start - nsslapd-connta…  …blesize

jchapma opened the pull request:
Description: Add a test case for issue 5956

Related: https://github.com/389ds/389-ds-base/issues/5956

Reviewed by:

jchapma committed:
Issue 5956 - After an upgrade the server won't start - nsslapd-conntablesize

Description: A commit, that increased the default connection table size included
the deprecation of the attribute nsslapd-conntablesize. An attempt to set the value
of this attribute issued a warning message and returned LDAP_OPERATIONS_ERROR. This
causes an issue after an upgrade, the server fails to start if nsslapd-conntablesize
is present in dse.ldif

Fix: Return LDAP_SUCESS instead...

Related: https://github.com/389ds/389-ds-base/issues/5956

Reviewed by:

jchapma committed:
Issue 5956 - After an upgrade the server won't start - nsslapd-conntablesize

Description: Add a test case for issue 5956

Related: https://github.com/389ds/389-ds-base/issues/5956

Reviewed by:

tbordaz commented on dirsrvtests/tests/suites/basic/basic_test.py:
```diff
@@ -1713,6 +1713,38 @@ def test_dscreate_with_different_rdn(dscreate_test_rdn_value):
+    assert dse_ldif.get(DN_CONFIG, attr)
+
+    inst.start()
+    assert inst.status() == True
```
You may check that the message "User setting of nsslapd-conntablesize is disabled,..." is logged in the error and after startup the server can answer simple request (like search base "")

jchapma replied:
Done

tbordaz commented on dirsrvtests/tests/suites/basic/basic_test.py:
```diff
@@ -1713,6 +1713,38 @@ def test_dscreate_with_different_rdn(dscreate_test_rdn_value):
         else:
             assert True
 
+def test_conntablesize_attr_dse(topo):
```
This test may be skipped on main and after 2.3

jchapma replied:
The test runs on branches 1.4.3 -> 2.2

jchapma committed:
basic_test.py - Improved test case
daemon.c - Tidy up get_connection_table_size
libglobs.c - Modified return value to trigger warning message
libglobs.c - Corrected use of SLAPD_DEFAULT_CONNTABLESIZE
slap.h - Corrected use of SLAPD_DEFAULT_CONNTABLESIZE

jchapma committed:
Merge branch '389-ds-base-1.4.3' into conntablesize-restart
